### DISCUSSION TITLE
How to select an adjacent sibling of an element with a specific class?

### DISCUSSION THREAD

**Discussion by MonaAghili** [tailwindlabs/tailwindcss#14124] [Help]
I want to select the sibling of an image, which is a `span` element with the classes `.indicator` and `.rounded-full`, and add a style to it.
This is my HTML:

```html
<div class="relative">
  <img src="../images/1.jpg" class="border border-[#707483] shadow-none" />
  <span class="indicator size-2 bg-[#004FC4] border-transparent block rounded-full border"></span>
</div>
```
I tried:

```html
<img src="../images/1.jpg" class="has-[+.indicator.rounded-full]:absolute has-[+.indicator.rounded-full]:top-0 has-[+.indicator.rounded-full]:left-0 has-[+.indicator.rounded-full]:translate-y-1/2 has-[+.indicator.rounded-full]:translate-x-1/2" />
```
but it doesn't seem to work.

**Comment by wongjn** [comment]
Seems to be working in [this Tailwind Play](https://play.tailwindcss.com/C3or6oGRC7) which would suggest it is something particular to your project. Consider providing an example that replicates your problem.

**Reply by MonaAghili** [reply]
Actually this doesn't work it seems apply style to image itself:
https://play.tailwindcss.com/MtKscDOEt8

**Reply by wongjn** [reply]
Yes, that is the behavior I would expect with the classes you have used. Do you intend for something else to happen?

**Reply by MonaAghili** [reply]
I want this style to apply to the `span`, not the image. When an image has a sibling with the `indicator` class, apply these styles to the sibling, similar to how the + selector works in CSS.
`img + .indicator {}`

**Reply by wongjn** [reply]
You could consider using [a `peer-*` variant](https://tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-sibling-state), like https://play.tailwindcss.com/1wxI9H23US
